### PR TITLE
[CELEBORN-228]Refactor PartitionFileSorter to avoid specific JDK dependency.

### PR DESCRIPTION
**Pull request by FMX**

### CODE REVIEW

File: worker/src/main/java/org/apache/celeborn/service/deploy/worker/storage/PartitionFilesSorter.java [hunk 1]
```diff
@@ -656,14 +646,68 @@ private void deleteOriginFiles() throws IOException {
       }
     }
 
-    private ByteBuffer expandBufferAndUpdateMemoryTracker(int oldCapacity, int newCapacity)
-        throws InterruptedException {
-      memoryManager.releaseSortMemory(oldCapacity);
-      memoryManager.reserveSortMemory(newCapacity);
-      while (!memoryManager.sortMemoryReady()) {
-        Thread.sleep(20);
+    protected void readStreamBySize(
```
Comment (waitinfuture) [inline worker/src/main/java/org/apache/celeborn/service/deploy/worker/storage/PartitionFilesSorter.java:649]: I think we don't need this for HDFS

**Review by waitinfuture — APPROVED** [review]
LGTM.

File: worker/src/main/java/org/apache/celeborn/service/deploy/worker/storage/PartitionFilesSorter.java [hunk 2]
```diff
@@ -656,14 +646,41 @@ private void deleteOriginFiles() throws IOException {
       }
     }
 
-    private ByteBuffer expandBufferAndUpdateMemoryTracker(int oldCapacity, int newCapacity)
-        throws InterruptedException {
-      memoryManager.releaseSortMemory(oldCapacity);
-      memoryManager.reserveSortMemory(newCapacity);
```
Comment (AngersZhuuuu) [inline worker/src/main/java/org/apache/celeborn/service/deploy/worker/storage/PartitionFilesSorter.java:662]: if newCapacity > maxSortMemoryRatio, will cause  `sortMemoryReady()` return false and this memory can't release, then cause this thread stuck here and cause PartitionSorter can't take FileSorter to executor.

**Review by AngersZhuuuu — COMMENTED** [review]
This pr fix below bug.

**Pull request merged**